From 59a9019a3c85cb8b8ac97fd82d21e1dd7a929f3b Mon Sep 17 00:00:00 2001
From: Hao Bui <hao.bui.yg@renesas.com>
Date: Tue, 26 Apr 2022 09:18:13 +0700
Subject: [PATCH 182/433] of: of_mdio: Correct loop scanning logic

The loop of PHY auto scanning should break once a succesfull
PHY registeration happens.

Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 drivers/of/of_mdio.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/of/of_mdio.c b/drivers/of/of_mdio.c
index 2b1ccb8..a58068e 100644
--- a/drivers/of/of_mdio.c
+++ b/drivers/of/of_mdio.c
@@ -178,8 +178,8 @@ int of_mdiobus_register(struct mii_bus *mdio, struct device_node *np)
 				 child->name, addr);
 
 			rc = of_mdiobus_register_phy(mdio, child, addr);
-			if (rc)
-				continue;
+			if (!rc)
+				break;
 		}
 	}
 
-- 
2.7.4

